client/src/app.tsx

```diff
@@ -130,18 +133,36 @@ class App extends React.Component<IAppProps, IAppState> {
                         <InfoPanel/>
                     </div>
                 )}
-                <div className="actions-panel-outer floating-panel">
-                    <div className="actions-panel-inner">
-                        <ActionsPanel
-                            orbitDistance={this.orbitDistanceSubject}
-                            islandState={this.state.islandState}
-                            location={this.perspectiveCamera.position}
-                        />
+                {this.state.islandState ? (
+                    <div className="actions-panel-outer floating-panel">
+                        <div className="actions-panel-inner">
+                            <ActionsPanel
+                                orbitDistance={this.orbitDistanceSubject}
+                                islandState={this.state.islandState}
+                                location={this.perspectiveCamera.position}
+                            />
+                        </div>
                     </div>
-                </div>
+                ) : (
+                    <h1>No island!</h1>
+                )}
             </div>
         )
     }
+
+    private fetchIsland(islandName: string): void {
+        this.props.storage.getIslandData(islandName).then(islandData => {
+            if (islandData) {
+                const island = new Island(
+                    this.islandSubject,
+                    islandData,
+                    this.fabricKernel,
+                    this.props.storage,
+                )
+                island.state.dispatch()
```

Comment:
What does `.dispatch()` do? Is it a RXjs thing?

Reply:
No, it calls next() on a Subject<IslandState>